content/concepts/multiplex/mplex.md

```diff
@@ -1,14 +1,48 @@
 ---
 title: "mplex"
-description: "mplex is a protocol developed for libp2p."
+description: "mplex is a stream multiplexer developed for libp2p."
 weight: 170
 ---
 
-mplex is a protocol developed for libp2p. The [spec](https://github.com/libp2p/specs/tree/master/mplex) defines a
-simple protocol for multiplexing that is widely supported across libp2p language implementations:
+## What is mplex?
 
-{{< alert icon="💡" context="note">}}
-This section is incomplete, and many of the articles are stubs. To help fill in
-the gaps, please see the issues linked in each article to add your input and
-help us prioritize the outstanding work.
+mplex is a stream multiplexer designed for libp2p.
+It is a simple protocol that does not provide many features offered by other
+stream multiplexers. Notably, mplex does not provide backpressure at the protocol
+level.
+
+mplex runs over a reliable, ordered pipe between two peers, such as a TCP socket
+or a Unix pipe. Peers can open, write to, close, and reset a stream by sending messages
+with the appropriate [flag value](https://github.com/libp2p/specs/tree/master/mplex#flag-values).
+
+Every message in mplex consists of a header and a length of prefixed data
+segment.
+
+### Drawbacks
+
+mplex does not support backpressure because it is designed to be more flexible and
+extensible than other multiplexing protocols. Instead of using a flow control mechanism
+like [yamux](yamux), mplex uses a message-based framing layer that enables it to multiplex
+different types of data streams, including stream-oriented data, datagrams, and other types
+of messages. This flexibility comes at the cost of some performance and reliability, as
+mplex is less efficient at transferring large amounts of data than other multiplexing
+protocols that support backpressure.
+
+Additionally, mplex has no limits on how many streams a peer can open. This allows mplex
+to support a wide range of applications and use cases, but it also means that mplex may
+not be as efficient as other multiplexing protocols that limit the number of streams that
+can be opened.
+
+{{< alert icon="" context="">}}
+**Yamux should be used over mplex in libp2p**. It is well-suited for applications that
+require fast, reliable data transfer, such as real-time communication or streaming media,
```

Comment:
My comment from the yamux document about real-time communication applies here too.